code/game/machinery/computer/arcade.dm

```diff
@@ -65,7 +65,19 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
 	// If it's a generic arcade machine, pick a random arcade
 	// circuit board for it and make the new machine
 	if(!circuit)
-		var/choice = pick(subtypesof(/obj/item/circuitboard/computer/arcade))
+		var/list/normalgames = list()
+		var/list/raregames = list()
+		for(var/games in subtypesof(/obj/item/circuitboard/computer/arcade))
+			var/obj/item/circuitboard/computer/arcade/thegame = games//you just lost
+			if(thegame.raregame)
+				raregames |= thegame
+			else
+				normalgames |= thegame
+		var/choice
+		if(prob(1))
+			choice = pick(raregames)
+		else
+			choice = pick(normalgames)
```

Comment:
just make a weighted light and use pickweight, this is gay

Reply:
Oh, good idea